src/rust/ide/lib/view/src/graph_editor.rs

```diff
@@ -1482,10 +1481,26 @@ fn new_graph_editor(app:&Application) -> GraphEditor {
     let sealed_outputs = outputs.seal(); // Done here to keep right eval order.
 
 
+
+    // ============================
+    // === Project Name Editing ===
+    // ============================
+
+
+    // === Commit project name edit ===
+
+    frp::extend! { network
+        eval_ touch.background.selected({
+            model.breadcrumbs.project_name.frp.commit.emit(())
+        });
+    }
```

Comment:
This is very, very, very bad! Every time I press in the background, the backend will get the info that the project was renamed! Please be very cautious about performance. It's ok to send this event here, but the FRP network of project name should handle it this way, that if you are not in the edit mode currently, nothing should happen. Of course "not in the edit mode" should be expressed in FRP network using gates etc.